src/main/java/com/example/bookstore/controller/BookController.java

```diff
@@ -47,4 +47,9 @@ public void delete(@PathVariable Long id) {
     public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
         return bookService.updateById(id, requestDto);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> searchBooks(dto.BookSearchParametersDto searchParameters) {

```

Comment:
`dto.BookSearchParametersDto` ?

this is weird :)

should be `BookSearchParametersDto`